go-controller/pkg/util/pod_annotation_unit_test.go

```diff
@@ -211,6 +218,20 @@ func TestUnmarshalPodAnnotation(t *testing.T) {
 			desc:        "verify successful unmarshal of pod annotation when *only* the MAC address is present",
 			inpAnnotMap: map[string]string{"k8s.ovn.org/pod-networks": `{"default":{"mac_address":"0a:58:fd:98:00:01"}}`},
 		},
+		{
```

Comment:
please add a test without join ips.

Reply:
this is the `UnmarshalPodAnnotation` one, it just have `k8s.ovn.org/pod-networks` and already have a test with just mac 
https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4878/files/88e1380294ad4b97a33e38e98291d10b29a36ad2#diff-8cc89a16276f003dd2f648bf85b5e23228e14807462ca836329d5159afcbe2d1L211

maybe you meant to add a test on `go-controller/pkg/allocator/pod/pod_annotation_test.go` for the new condition ?

Reply:
My bad; yes, add it where you're saying.

It's a bit more complex, but you got it.

Reply:
I tried, i really think it is overkill for `go-controller/pkg/allocator/pod/pod_annotation_test.go`
it is too far and small / deep for such test
moreover no where in the code there is such case where `OVNNodeGRLRPAddrs` annotation has a network without ips (function is guarded that network do exists)
Please reconsider that if i may ask, another option is to wrap 
https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4878/files#diff-3d3fe81a31268662d021e60eb37f4a1f7e96ddb7e61de7f49acbbd2ae82e81caR596-R602
in a function and test it, but even that imo is too much

Reply:
if i set
```
ifaddrs = `{"network":{}}`
```
Then even without the new guard
we fail on 
https://github.com/ovn-kubernetes/ovn-kubernetes/blob/6c366bb49d6fa57c80d6b39834952ce1d4d9dbb6/go-controller/pkg/util/node_annotations.go#L700

is it fine to remove the guard / test then or you still want it one of them / both ?